### PR TITLE
Feature/api azure

### PR DESCRIPTION
**Pull request by DzmitryI**

### CODE REVIEW

File: src/components/UI/bodyTable/BodyTable.js [hunk 1]
```diff
@@ -2,14 +2,14 @@ import React from 'react';
...
-  arr.map((arr, index) => {
+  arr.map((arrTd, index) => {
     const renderItems = (bodies) =>
       bodies.map(({ className, value }, index) => (
         <td key={className + index} className={className}>
           {value}
         </td>
       ));
-    return <tr key={index}>{renderItems(arr)}</tr>;
+    return <tr key={arrTd + index}>{renderItems(arrTd)}</tr>;
```
Comment (maximsan) [inline src/components/UI/bodyTable/BodyTable.js:12]: I guess arrTd is enough

File: package.json [hunk 2]
```diff
@@ -52,19 +55,18 @@
...
     "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-prettier": "^3.1.3",
     "eslint-plugin-react": "^7.19.0",
-    "eslint-plugin-react-hooks": "^3.0.0",
+    "eslint-plugin-react-hooks": "^2.5.1",
     "husky": "^4.2.5",
     "prettier": "^2.0.4",
     "pretty-quick": "^2.0.1",
-    "react-test-renderer": "^16.13.1"
+    "typescript": "^3.8.3"
```
Comment (maximsan) [inline package.json:70]: do you really need it ?

File: package.json [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
   "name": "dev-incubator-management-system",
-  "version": "0.1.0",
+  "version": "1.0.0",
   "description": "It is a system for getting tasks and tracking time.",
   "author": "Dmitriy Ivanov",
   "license": "MIT",
-  "homepage": "https://github.com/DzmitryI/DIMS.UI-3",
+  "engines": {
+    "node": "12.16.x",
+    "npm": "6.14.x"
```
Comment (maximsan) [inline package.json:9]: I believe you know that these are rules that more necessary for libraries :)

File: src/components/UI/footer/Footer.js [hunk 4]
```diff
@@ -12,9 +12,7 @@ const Footer = ({ theme }) => {
         <SocialLink href='mailto:dmitriy.ivanou@gmail.com' src={imgMail} alt='mail icon' />
         <SocialLink href='https://github.com/DzmitryI' src={imgGit} alt='git icon' />
       </div>
-      <span>
-        <sup>&copy;</sup> 2020 Devincubator.
-      </span>
+      <span>&copy; 2020 Dev Incubator.</span>
```
Comment (maximsan) [inline src/components/UI/footer/Footer.js:15]: add function for dynamic year and unit test for it

File: src/components/UI/header/Header.js [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 import { logout } from '../../../store/actions/auth';
 import { withTheme } from '../../../hoc';
 import imgMain from '../../../assets/images/main.png';
-import PropTypes from 'prop-types';
 
 const Header = (props) => {
-  const { isAuthenticated, email, theme } = props;
-  let links = [];
+  const { isAuthenticated, email, theme, logout } = props;
+  let arrLinks = [];
```
Comment (maximsan) [inline src/components/UI/header/Header.js:15]: use const

File: src/components/membersGrid/MembersGrid.js [hunk 6]
```diff
@@ -65,9 +58,16 @@ class MembersGrid extends Component {
     this.props.onProgressClick(memberId, name);
   };
 
+  countAge = (value) => {
+    const curDate = new Date();
+    const birthDate = new Date(value);
+    const age = curDate.getFullYear() - birthDate.getFullYear();
+    return curDate.setFullYear(curDate.getFullYear()) < birthDate.setFullYear(curDate.getFullYear()) ? age - 1 : age;
+  };
```
Comment (maximsan) [inline src/components/membersGrid/MembersGrid.js:66]: move to helpers and add unit tests

File: src/components/membersGrid/MembersGrid.js [hunk 7]
```diff
@@ -76,7 +76,7 @@ class MembersGrid extends Component {
           <Cell value={`${!!curDirect && curDirect.name}`} />
           <Cell value={`${education}`} />
           <Cell value={getDate(startDate)} />
-          <Cell value={`${this.countAge(birthDate)}`} />
+          <Cell value={(birthDate && `${this.countAge(birthDate)}`) || age} />
```
Comment (maximsan) [inline src/components/membersGrid/MembersGrid.js:79]: do you need `` here ?

**Comment by maximsan** [comment]
Please resolve conflicts and leave **only 1 lock file!** It's very important, bu careful!

File: package.json (hunk 2 above)
Comment (DzmitryI) [reply package.json:70]: fixed

File: src/components/UI/bodyTable/BodyTable.js (hunk 1 above)
Comment (DzmitryI) [reply src/components/UI/bodyTable/BodyTable.js:12]: fixed

File: package.json (hunk 3 above)
Comment (DzmitryI) [reply package.json:9]: fixed

File: src/components/UI/footer/Footer.js (hunk 4 above)
Comment (DzmitryI) [reply src/components/UI/footer/Footer.js:15]: fixed

File: src/components/membersGrid/MembersGrid.js (hunk 6 above)
Comment (DzmitryI) [reply src/components/membersGrid/MembersGrid.js:66]: fixed

File: src/components/membersGrid/MembersGrid.js (hunk 7 above)
Comment (DzmitryI) [reply src/components/membersGrid/MembersGrid.js:79]: fixed

**Pull request closed without merging**